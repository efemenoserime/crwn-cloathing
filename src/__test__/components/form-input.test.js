/* eslint-disable testing-library/await-async-query */
import React from "react";
import { cleanup, render, screen } from "@testing-library/react";

import FormInput from "../../components/form-input/form-input.component";

afterEach(cleanup);

describe("<FormInput />", () => {
	const mockHandleChange = jest.fn();

	it("renders component successfully", () => {
		const { asFragment } = render(<FormInput />);
		expect(asFragment(<FormInput />)).toMatchSnapshot();
	});

	it("renders form-input without label", () => {
		expect.assertions(1);
		render(<FormInput handleChange={mockHandleChange} />);
		return screen.findByTestId("form-input-title").catch((error) => {
			expect(error).toBeTruthy();
		});
	});

	it("renders form-input with label", () => {
		expect.assertions(1);
		const mockLabel = "Full Name";
		render(<FormInput handleChange={mockHandleChange} label={mockLabel} />);
		return screen.findByTestId("form-input-title").then((element) => {
			expect(element.textContent).toEqual(mockLabel);
		});
	});
});
