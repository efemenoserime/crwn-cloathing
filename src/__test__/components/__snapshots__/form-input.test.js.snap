// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> renders component successfully 1`] = `
<DocumentFragment>
  <div
    class="group"
  >
    <input
      class="form-input"
    />
  </div>
</DocumentFragment>
`;
